listenbrainz/mbid_mapping_writer/mbid_mapper.py

```diff
@@ -0,0 +1,284 @@
+        """
+            Evaluate the given prepared search terms and hit. If the hit doesn't match,
+            attempt to detune it and try again for detuned artist and detuned recording.
+            If the hit is good enough, return it, otherwise return None.
+        """
+        ac_hit = hit['document']['artist_credit_name']
+        r_hit = hit['document']['recording_name']
+
+        ac_detuned = self.detune_query_string(ac_hit, True)
+        r_detuned = self.detune_query_string(r_hit, False)
+
+        while True:
+            ac_dist, r_dist = self.compare(
+                artist_credit_name, recording_name, prepare_query(ac_hit), prepare_query(r_hit))
+
+            # If we detuned one or more fields and it matches, the best it can get is a low quality match
+            if (self.is_ac_detuned or self.is_r_detuned) and \
+                    ac_dist <= self.MATCH_TYPE_MED_QUALITY_MAX_EDIT_DISTANCE and \
+                    r_dist <= self.MATCH_TYPE_MED_QUALITY_MAX_EDIT_DISTANCE:
+                self._log(Markup(f"""<b>low quality</b>: ac_detuned {self.is_ac_detuned}, r_detuned {self.is_r_detuned},
+                                     ac_dist {ac_dist:.3f} r_dist {r_dist:.3f}"""))
+                return (hit, MATCH_TYPE_LOW_QUALITY)
+
+            # For exact matches, return exact match. duh.
+            if ac_dist == 0 and r_dist == 0:
+                self._log(Markup(f"""<b>exact_match</b>: ac_detuned {self.is_ac_detuned}, r_detuned {self.is_r_detuned}"""))
+                return (hit, MATCH_TYPE_EXACT_MATCH)
+
+            # If both fields are above the high quality threshold, call it high quality
+            if ac_dist <= self.MATCH_TYPE_HIGH_QUALITY_MAX_EDIT_DISTANCE and \
+               r_dist <= self.MATCH_TYPE_HIGH_QUALITY_MAX_EDIT_DISTANCE:
+                self._log(Markup(f"""<b>high quality</b>: ac_detuned {self.is_ac_detuned}, r_detuned {self.is_r_detuned},
+                                     ac_dist {ac_dist:.3f} r_dist {r_dist:.3f}"""))
+                return (hit, MATCH_TYPE_HIGH_QUALITY)
+
+            # If both fields are above the medium quality threshold, call it medium quality
+            if ac_dist <= self.MATCH_TYPE_MED_QUALITY_MAX_EDIT_DISTANCE and \
+                    r_dist <= self.MATCH_TYPE_MED_QUALITY_MAX_EDIT_DISTANCE:
+                self._log(Markup(f"""<b>med quality</b>: ac_detuned {self.is_ac_detuned}, r_detuned {self.is_r_detuned},
+                                     ac_dist {ac_dist:.3f} r_dist {r_dist:.3f}"""))
+                return (hit, MATCH_TYPE_MED_QUALITY)
+
+            # Poor results so far, lets try detuning fields and try again
+            if ac_dist > self.MATCH_TYPE_MED_QUALITY_MAX_EDIT_DISTANCE and ac_detuned:
+                ac_hit = ac_detuned
+                ac_detuned = ""
+                self.is_ac_detuned = True
+                self._log(Markup(f"""<b>no match</b>, ac_dist too high {ac_dist:.3f} continuing: detune ac"""))
```

Comment:
Since its possible to detune both the search artist credit name and the hit's artist credit name, we should mention in the logged line that its `ac_hit` that has been detuned. Similarly, `r_hit` in next case.